src/patternfly/components/Nav/nav.scss

```diff
@@ -548,157 +372,91 @@
-    &.pf-m-focus,
-    &:focus {
-      --pf-c-nav__tertiary-list-link--Color: var(--pf-c-nav__tertiary-list-link--focus--Color);
-
-      &::after {
-        background-color: var(--pf-c-nav__tertiary-list-link--focus--after--BackgroundColor);
-      }
-    }
-
-    &.pf-m-active,
-    &:active {
-      --pf-c-nav__tertiary-list-link--Color: var(--pf-c-nav__tertiary-list-link--active--Color);
-
-      &::after {
-        background-color: var(--pf-c-nav__tertiary-list-link--active--after--BackgroundColor);
-      }
-    }
-
-    &.pf-m-current {
-      --pf-c-nav__tertiary-list-link--Color: var(--pf-c-nav__tertiary-list-link--m-current--Color);
-
-      &::after {
-        background-color: var(--pf-c-nav__tertiary-list-link--m-current--after--BackgroundColor);
-      }
-    }
-
-    // After - border bottom treatment
-    &::after {
-      position: absolute;
-      bottom: 0;
-      left: 0;
-      display: block;
-      width: 100%;
-      height: var(--pf-c-nav__tertiary-list-link--after--Height);
-      content: "";
-    }
+  .pf-c-nav__link:hover,
+  .pf-c-nav__link.pf-m-current {
+    --pf-c-nav__link--after--BorderBottomWidth: var(--pf-c-nav__tertiary-list__link--after--BorderWidth);
   }
 }
 
 // Subnav
 .pf-c-nav__subnav {
-  flex: 1 0 100%;
+  --pf-c-nav__link--before--BorderTopWidth: var(--pf-c-nav__subnav__link--before--BorderTopWidth);
+  --pf-c-nav__link--before--BorderBottomWidth: var(--pf-c-nav__subnav__link--before--BorderBottomWidth);
+  --pf-c-nav__link--FontSize: var(--pf-c-nav__subnav__link--FontSize);
+  --pf-c-nav__simple-list__link--MarginTop: var(--pf-c-nav--subnav__simple-list__link--MarginTop);
+  --pf-c-nav__simple-list__link--MarginBottom: var(--pf-c-nav--subnav__simple-list__link--MarginBottom);
+  --pf-c-nav__simple-list__link--FontSize: var(--pf-c-nav__subnav__link--FontSize);
 
-  // Set default states
   max-height: 0;
-  margin-top: 0;
-  overflow: hidden;
-  opacity: 0;
+  padding-bottom: var(--pf-c-nav__subnav--PaddingBottom);
+  padding-left: var(--pf-c-nav__subnav--PaddingLeft);
   transition: var(--pf-c-nav--Transition);
 
-  // Hide scrollbars
-  &::-webkit-scrollbar {
-    display: none;
-    visibility: hidden;
+  .pf-c-nav__item:last-child {
+    --pf-c-nav__link--before--BorderBottomWidth: 0;
   }
 
-  // Expanded list item
   .pf-c-nav__item.pf-m-expanded & {
     max-height: var(--pf-c-nav__subnav--MaxHeight);
-    margin-top: var(--pf-c-nav__subnav--MarginTop);
     overflow-y: auto;
     opacity: 1;
   }
+
+  .pf-c-nav__link:not(.pf-m-current):hover {
+    --pf-c-nav__link--after--BorderLeftWidth: var(--pf-c-nav__subnav__link--not--m-current--hover--BorderLeftWidth);
+    --pf-c-nav__link--after--BorderColor: var(--pf-c-nav__subnav__link--not--m-current--hover--BorderColor);
+    --pf-c-nav__link--after--BorderTopWidth: 0;
+  }
+
+  &::-webkit-scrollbar {
+    display: none;
+    visibility: hidden;
+  }
 }
 
+// Section
 .pf-c-nav__section {
   margin-top: var(--pf-c-nav__section--MarginTop);
 
```

Comment:
It's not in this changeset, but small nit - the line below this can be written as `+ &`